tests/api_tests/base/test_filters.py

```diff
@@ -0,0 +1,228 @@
+            elif match['op'] == 'icontains':
+                assert_equal(match['value'], 'bar')
+            else:
+                self.fail()
+
+    def test_parse_query_params_raises_InvalidFilterError_bad_field(self):
+        query_params = {
+            'filter[fake]': 'foo'
+        }
+        with assert_raises(InvalidFilterError):
+            self.view.parse_query_params(query_params)
+
+    def test_parse_query_params_raises_InvalidFilterComparisonType(self):
+        query_params = {
+            'filter[string_field][gt]': 'foo'
+        }
+        with assert_raises(InvalidFilterComparisonType):
+            self.view.parse_query_params(query_params)
+
+    def test_parse_query_params_raises_InvalidFilterMatchType(self):
+        query_params = {
+            'filter[date_field][icontains]': '2015'
+        }
+        with assert_raises(InvalidFilterMatchType):
+            self.view.parse_query_params(query_params)
+
+    def test_parse_query_params_raises_InvalidFilterOperator(self):
+        query_params = {
+            'filter[int_field][bar]': 42
+        }
+        with assert_raises(InvalidFilterOperator):
+            self.view.parse_query_params(query_params)
+
+    def test_InvalidFilterOperator_parameterizes_valid_operators(self):
+        query_params = {
+            'filter[int_field][bar]': 42
+        }
+        try:
+            self.view.parse_query_params(query_params)
+        except InvalidFilterOperator as err:
+            ops = re.search(r'one of (?P<ops>.+)\.$', err.detail).groupdict()['ops']
+            assert_equal(ops, "gt, gte, lt, lte, eq")
+
+        query_params = {
+            'filter[string_field][bar]': 'foo'
+        }
+        try:
+            self.view.parse_query_params(query_params)
+        except InvalidFilterOperator as err:
+            ops = re.search(r'one of (?P<ops>.+)\.$', err.detail).groupdict()['ops']
+            assert_equal(ops, "contains, icontains, eq")
+
+
+    def test_parse_query_params_supports_multiple_filters(self):
+        query_params = {
+            'filter[string_field]': 'foo',
+            'filter[string_field]': 'bar',
+        }
+
+        fields = self.view.parse_query_params(query_params)
+        assert_in('string_field', fields)
+        for match in fields['string_field']:
+            assert_in(match['value'], ('foo', 'bar'))
+
+    def test_convert_value_bool(self):
+        value = 'true'
+        field = FakeSerializer._declared_fields['bool_field']
+        value = self.view.convert_value(value, field)
+        assert_true(isinstance(value, bool))
+        assert_true(value)
+
+    def test_convert_value_date(self):
+        value = '2014-12-12'        
+        field = FakeSerializer._declared_fields['date_field']
+        value = self.view.convert_value(value, field)
+        assert_true(isinstance(value, datetime.datetime))
+        assert_equal(value, parser.parse('2014-12-12'))
+
+    def test_convert_value_number(self):
```

Comment:
The two halves of this test are separate things; please separate them into separate tests. (so that if the first one fails we are still testing the other as well)